keps/1618-optional-gc-of-workloads/README.md

```diff
@@ -0,0 +1,441 @@
+
+#### Integration tests
+
+<!--
+Describe what tests will be added to ensure proper quality of the enhancement.
+
+After the implementation PR is merged, add the names of the tests here.
+-->
+
+- TBA
+
+
+[//]: # (### Graduation Criteria)
+
+[//]: # ()
+[//]: # (<!--)
+
+[//]: # ()
+[//]: # (Clearly define what it means for the feature to be implemented and)
+
+[//]: # (considered stable.)
+
+[//]: # ()
+[//]: # (If the feature you are introducing has high complexity, consider adding graduation)
+
+[//]: # (milestones with these graduation criteria:)
+
+[//]: # (- [Maturity levels &#40;`alpha`, `beta`, `stable`&#41;][maturity-levels])
+
+[//]: # (- [Feature gate][feature gate] lifecycle)
+
+[//]: # (- [Deprecation policy][deprecation-policy])
+
+[//]: # ()
+[//]: # ([feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md)
+
+[//]: # ([maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions)
+
+[//]: # ([deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+
+[//]: # (-->)
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+- Initial PR created for the feature pre-KEP [#2686](https://github.com/kubernetes-sigs/kueue/pull/2686).
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+- Deleting objects is inherently complex, and garbage collection (GC) behavior can often be 
+surprising for end-users who did not configure the service themselves. Even though the 
+mechanism proposed in this KEP is relatively simple and must be explicitly enabled, 
+it is a global mechanism applied at the controller level. The author has added some 
+observability to the controller itself (logging and emitting deletion events), 
+but Kubernetes users and developers accustomed to tracking object lifecycles 
+using per-object policies might be surprised by this behavior.
+
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+- Leave the current state of things as is, not implementing any garbage 
+collection mechanism for finished Workloads.
+- Implement retention policies attached per object (like `.spec.ttlSecondsAfterFinished`) instead of having 
```

Comment:
Also, workloads are objects internal to Kueue and created by Kueue, so it is not clear how users could set `.spec.ttlSecondsAfterFinished` on a workload.

Reply:
Good point. I forgot that the spec section is not populated by Kueue but by the user. I removed references to that field.